### PR TITLE
Update InstallAppRequest type to new format

### PR DESCRIPTION
**Pull request by matthme**
This PR contains the type changes required in order to adhere to https://github.com/holochain/holochain/pull/4416 and its follow-up PR https://github.com/holochain/holochain/pull/4488

### CODE REVIEW

**Review by jost-s — CHANGES_REQUESTED** [review]
Please add a test or modify an existing one that uses roles_settings.

**Comment by matthme** [comment]
I'm confused - why do the tests even pass? Are the tests not actually ran in CI?

**Comment by jost-s** [comment]
The tests pass because `memproofs` is just ignored and `roles_settings` is optional.

**Comment by matthme** [comment]
Test added.

**Comment by matthme** [comment]
Hmm...actually the problem that `Uint8Array`s don't deserialize to `YamlProperties` is a problem. This means that you can't even pass in agent public keys there unless you base64 encode them or conver them to an `Array<number>`.

File: test/e2e/index.ts [hunk 1]
```diff
@@ -562,6 +564,77 @@ test(
...
+        },
+        resources: {},
+      },
+      roles_settings: {
+        foo: {
+          type: "Provisioned",
+          membrane_proof: new Uint8Array(6),
+          modifiers: {
+            network_seed: "hello",
+            properties: yaml.dump({ progenitor: progenitorKey }),
```
Comment (matthme) [inline test/e2e/index.ts:613]: Changed the test now to use a yaml string.
cc @ThetaSinner

**Comment by matthme** [comment]
Okay never mind. I think this is totally appropriate and the app developer can be expected to take care of any required conversion for example using `js-yaml` (we just need to make sure that the js-client actually returns a useful error which is not currently the case https://github.com/holochain/holochain-client-js/issues/299).

File: package.json [hunk 2]
```diff
@@ -48,13 +48,15 @@
     "isomorphic-ws": "^5.0.0",
     "js-base64": "^3.7.5",
     "js-sha512": "^0.9.0",
+    "js-yaml": "4.1.0",
```
Comment (jost-s) [inline package.json]: This can be a dev dependency too.

File: src/api/admin/types.ts [hunk 3]
```diff
@@ -156,7 +156,43 @@ export type MembraneProof = Uint8Array;
...
+      type: "Provisioned";
+      membrane_proof?: MembraneProof;
+      modifiers?: DnaModifiersOpt;
+    };
+
+/**
+ *  @public
+ * Any value that is serializable to a Yaml value
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (jost-s) [inline src/api/admin/types.ts]: Is this needed here? It should only produce a warning.
Actually `unknown` would be the idiomatic type.

File: test/e2e/common.ts [hunk 4]
```diff
@@ -117,7 +117,7 @@ export const installAppAndDna = async (
     installed_app_id,
     agent_key: agent,
     path,
-    membrane_proofs: {},
+    roles_settings: {},
```
Comment (jost-s) [inline test/e2e/common.ts]: `roles_settings` is optional
```suggestion
```

File: test/e2e/common.ts [hunk 5]
```diff
@@ -159,7 +159,7 @@ export const createAppWsAndInstallApp = async (
...
```
Comment (jost-s) [inline test/e2e/common.ts]:
```suggestion
```

File: test/e2e/index.ts [hunk 6]
```diff
@@ -85,7 +87,7 @@ test(
       installed_app_id,
       agent_key,
       path: `${FIXTURE_PATH}/test.happ`,
-      membrane_proofs: {},
+      roles_settings: {},
```
Comment (jost-s) [inline test/e2e/index.ts]: and so on further down
```suggestion
```

File: test/e2e/index.ts [hunk 7]
```diff
@@ -1046,7 +1119,7 @@ test(
 
     const tag = "test_tag";
     const link: Link = await client.callZome({
-      cap_secret: null,
+      cap_secret: undefined,
```
Comment (jost-s) [inline test/e2e/index.ts]:
```suggestion
```

**Review by jost-s — CHANGES_REQUESTED** [review]
LGTM
a suggestion for the yaml properties type and some cleanup

**Comment by matthme** [comment]
Thank you! Addressed all of these now.

File: Cargo.toml [hunk 8]
```diff
@@ -7,4 +7,4 @@ resolver = "2"
 opt-level = "z"
 
 [workspace.dependencies]
-hdk = "0.5.0-dev.0"
+hdk = "0.5.0-dev.6"
```
Comment (jost-s) [inline Cargo.toml]: This is intentionally kept at `.0` as long as there are no breaking changes in the HDK.

File: test/e2e/common.ts (hunk 5 above)
Comment (jost-s) [reply test/e2e/common.ts]: I'm still here =)

**Review by jost-s — CHANGES_REQUESTED** [review]
Just a few more bits

File: Cargo.toml (hunk 8 above)
Comment (matthme) [reply Cargo.toml]: Alright!

**Comment by matthme** [comment]
search and replace for the win.

File: test/e2e/common.ts (hunk 5 above)
Comment (jost-s) [reply test/e2e/common.ts]: And still here

File: test/e2e/common.ts (hunk 4 above)
Comment (jost-s) [reply test/e2e/common.ts]: Delete me

**Review by jost-s — CHANGES_REQUESTED** [review]

File: test/e2e/common.ts (hunk 4 above)
Comment (matthme) [reply test/e2e/common.ts]: done.

**Review by jost-s — APPROVED** [review]

**Pull request merged**